openquake/hazardlib/shakemap/parsers.py

```diff
@@ -472,17 +477,34 @@ def load_rupdic_from_finite_fault(usgs_id, mag, products):
     """
     Extract the finite fault properties from products.
     NB: if the finite-fault list contains multiple elements we take the
-    first one.
+    preferred one.
     """
     logging.info('Getting finite-fault properties')
-    try:
-        ff = products['finite-fault']
-    except KeyError:
+    if 'finite-fault' not in products:
+        # e.g. us6000phrk
         # FIXME: not tested
-        raise MissingLink('There is no finite-fault info for %s' % usgs_id)
-    if isinstance(ff, list):
-        ff = ff[0]
+        err_msg = f'There is no finite-fault info for {usgs_id}'
+        err = {"status": "failed", "error_msg": err_msg}
+        return None, err
+    ffs = products['finite-fault']
+    ff = _get_preferred_item(ffs)
     p = ff['properties']
+    # TODO: we probably need to get the rupture coordinates from shakemap_polygon.txt
```

Comment:
Anirudh says
> the downloadable rupture polygon files are not very reliable or consistent in format. For instance, the finite fault polygon file for the 2024 Hualien earthquake does contain the rupture coordinates: https://earthquake.usgs.gov/product/finite-fault/us7000m9g4_1/us/1712690873099/shakemap_polygon.txt, but the 2023 Kahramanmaraş finite fault polygon files are empty: https://earthquake.usgs.gov/product/finite-fault/us6000jllz_1/us/1676951251912/shakemap_polygon.txt and https://earthquake.usgs.gov/product/finite-fault/us6000jlqa_1/us/1676676771092/shakemap_polygon.txt
> On the other hand, the rupture.json file from the ShakeMap downloads page is reliable and consistent

so we should never read the polygon and you can remove the commented code below.